src/HypothesisClient/Credentials/JWTSigningCredentials.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace eLife\HypothesisClient\Credentials;
+
+use eLife\HypothesisClient\Clock\Clock;
+use Firebase\JWT\JWT;
+
+class JWTSigningCredentials extends Credentials
+{
+    private $clock;
+    private $expire;
+
+    public function __construct(string $clientId, string $clientSecret, string $authority, Clock $clock, int $expire = 600)
+    {
+        parent::__construct($clientId, $clientSecret, $authority);
+        $this->clock = $clock;
+        $this->expire = $expire;
+    }
+
+    public function getExpire() : int
```

Comment:
Without seeing the intended usage I can't be sure of this, but `getExpire()` and `getStartTime()` seem two getters that are not used outside of the tests for this class. If the goal is to only call `getJWT` from client code, these two can be removed (and hence their tests too). The advantage is less code to read and maintain, fewer tests, same functionality.

Reply:
(the inherited `getAuthority()` _is_ stricly needed instead)